_posts/2024-12-05-dev-api.md

```diff
@@ -0,0 +1,142 @@
+were not sure about their design. In the 1.6 release, new tags are introduced and
+using them looks like the following:
+
+```python
+from sklearn.base import BaseEstimator, ClassifierMixin
+
+class MyEstimator(ClassifierMixin, BaseEstimator):
+
+  ...
+
+  def __sklearn_tags__(self):
+    tags = super().__sklearn_tags__()
+    # modify tags here
+    tags.non_deterministic = True
+    return tags
+```
+
+The new tags mostly follow the same structure as the old tags, but there are certain
+changes to them. The main change is that the old `_xfail_checks` is no more present
+in the new tags. That tag was used to tell the common testing tools about the tests
+which are known to fail and are to be skipped. That information is now directly passed
+to the test functionalities. The old way of skipping a test was the following:
+
+```python
+from sklearn.base import BaseEstimator, ClassifierMixin
+
+class MyEstimator(ClassifierMixin, BaseEstimator):
+
+  ...
+
+  def _more_tags(self):
+    return {
+      "_xfail_checks": {
+        "check_to_skip_name": "this check is known to fail",
+        ...
+      }
+    }
+```
+
+And then when calling `check_estimator` or using `parametrize_with_checks` with `pytest`
+would automatically ignore those tests for the estimator.
+
+Instead, in this release, you pass that information directly to those methods:
+
+```python
+from sklearn.utils.estimator_checks import check_estimator, parametrize_with_checks
+
+CHECKS_EXPECTED_TO_FAIL = {
+  "check_to_skip_name": "this check is known to fail",
+  ...
+}
+
+# Using check_estimator
+def test_with_check_estimator():
+  check_estimator(MyEstimator(), expected_failed_checks=CHECKS_EXPECTED_TO_FAIL)
+
+# Using parametrize_with_checks
+@parametrize_with_checks(
+  [MyEstimator()],
+  expected_failed_checks=lambda est: CHECKS_EXPECTED_TO_FAIL
+)
+def test_with_parametrize_with_checks(estimator, check):
+  check(estimator)
+```
+
+While working on the testing infrastructure, we have also been working on improving our
+tests and that means in this release we had a particularly higher number of changes in
+their names and what they do. The changes should have made it easier for developers to
+fix issues with their estimators. Note that you can now pass `legacy=False` to both
+`check_estimator` and `parametrize_with_checks` to include only strictly API related
+tests.
+
+The above changes means developers need to update their estimators and depending on
+what they use, write scikit-learn version specific code to handle supporting multiple
+scikit-learn versions. To make that process easier, we've worked on a package called
+[`sklearn_compat`](https://github.com/sklearn-compat/sklearn-compat/). You can either
+depend on it as a package dependency, or vendor a single file inside your project. At
+the moment this project is in its infancy and might change in the future. But hopefully
+it helps developers out there.
```

Comment:
Maybe add that we would be happy to get developer feedback and are open to cover some use-cases that we did not yet thought about.

Reply:
Ups this is writen in the next line :)